pkg/apis/smith/v1/types.go

```diff
@@ -194,11 +197,27 @@ type Resource struct {
 	Name ResourceName `json:"name"`
 
 	// Explicit dependencies.
-	DependsOn []ResourceName `json:"dependsOn,omitempty"`
+	References []Reference `json:"reference,omitempty"`
 
 	Spec ResourceSpec `json:"spec"`
 }
 
+// +k8s:deepcopy-gen=true
+// Refer to a part of another object
+type Reference struct {
+	Name     ReferenceName `json:"name,omitempty"`
```

Comment:
What is the benefit of having `name` additionally to `resource`? Will we ever declare dependencies on the same resource more than once? I guess not. And `resource` is unique within a bundle, so it seems like just a way to shorten the potentially long resource names to make it nicer?